### DISCUSSION TITLE
Define the text color for a TextTheme?

### DISCUSSION THREAD

**Discussion by stefandevo** [rydmike/flex_color_scheme#63] [Q&A] [answered]
Wondering what the best solution would be if you define a custom `TextTheme` but want to set a specific TextColor for .light and .dark styles?

if `blendTextTheme: false,` set it reverts to default styling (where the couple of first heading colors are grayish and the rest is black). Removing this, will use the primary color defined for all text theme styles.

So I came up with two TextThemes; one for light and one for dark, where only the `color` property is different.

Any suggestions?

**Comment by stefandevo** [comment, marked as answer]
Found the answer :-)

I can use the `.apply` on the `TextTheme` class to set the color.

**Reply by rydmike** [reply]
Yes apply is the way to go in order to modify color on the TextTheme. 

## Some more font coloring nice to know info.

The "grayish" default fonts in Flutter are actually using opacity on the white and black types. For some types it is important to keep same or similar level of opacity on the global default TextTheme for each style. Material and Flutter uses opacity of some style, to in some usage situations, eg in ListTile defaults, that copies opacity from a style that has it to make the used style of subtitle partially transparent and grey for less impact. If you don't do this, ListTile will nor look right when it uses its default style.

Because of this, the slightly primary tinted font option offered in FlexColorScheme (FCS), retains some transparency on same styles. But using a bit less opacity, while also using alpha blend to mix in a hint of whatever primary color is used. It is quite subtle though.

Using a primary color tint on the font only works well on neutral backgrounds, or if they have a light tint of primary color too. If text is used a lot on some other background color than neutral or primary tinted (M3 backgrounds are very lightly primary tinted too, like background blends in FCS), then just going with the plain white/black default font where some styles have opacity is a better choice. This is mentioned in API docs too. In most normal use though, the color tinted TexTheme can be nice subtle additional effect. Maybe in some future version I'll add some props to allow tuning its strength.
